src/nnet3/nnet-nnet.h

```diff
@@ -130,6 +130,12 @@ class Nnet {
   /// caller.
   const Component *GetComponent(int32 c) const;
 
+  /// Replace the component indexed by c with a new component.
+  /// Frees previous component indexed by c.
+  /// Changing the identity of a component is dangerous; this was made
+  /// only to replaced RepeatedAffine components with BlockAffine components,
```

Comment:
You don't have to mention the reason it was added- it's a reasonable thing to have there regardless.